### PR TITLE
grep exercise

### PR DESCRIPTION
**Pull request by cezarykosko**
calling dibs on the grep exercise

### CODE REVIEW

**Comment by parkerl** [comment]
Hi @cezarykosko how is this coming along?

**Comment by cezarykosko** [comment]
Hey,
Sorry for the delay, had a preliminary version ready way back, but managed to get it close to finished but today (was busy with my BSc thesis defence :))

**Comment by parkerl** [comment]
No worries! Hope that goes/went well!

File: exercises/grep/grep.exs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+defmodule Grep do
+  def grep(pattern, flags, files) do
+    raise "Your implementation goes here"
```
Comment (parkerl) [inline exercises/grep/grep.exs]: Can you add a typespec and remove the raise please?

File: exercises/grep/grep_test.exs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        File.rm!("iliad.txt")
+        File.rm!("midsummer-night.txt")
+        File.rm!("paradise-lost.txt")
+      end
+    end
+  end
+
+  @moduletag :files
+
+
```
Comment (parkerl) [inline exercises/grep/grep_test.exs]: remove extra line

File: exercises/grep/grep_test.exs [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  describe "Test grepping a single file" do
```
Comment (parkerl) [inline exercises/grep/grep_test.exs]: "Test grepping a single file" -> "grepping a single file"

Comment (parkerl) [inline exercises/grep/grep_test.exs]: Also add a line between the `describe` and the first `test` please.

File: exercises/grep/grep_test.exs [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        """
+    end
+
+    @tag :pending
+    test "no matches, various flags" do
+      assert Grep.grep("Gandalf", ["-n", "-l", "-x", "-i"], ["iliad.txt"]) == ""
+    end
+  end
+
+  describe "Test grepping multiples files at once" do
```
Comment (parkerl) [inline exercises/grep/grep_test.exs]: "Test grepping multiples files at once" -> "grepping multiples files at once"

**Comment by cezarykosko** [comment]
Done :)

**Comment by parkerl** [comment]
Looks 👍 to me... @devonestes ?

File: exercises/grep/grep_test.exs [hunk 5]
```diff
@@ -0,0 +1,259 @@
+if !System.get_env("EXERCISM_TEST_EXAMPLES") do
+  Code.load_file("bob.exs", __DIR__)
```
Comment (devonestes) [inline exercises/grep/grep_test.exs]: I think we're loading the wrong file here - should probably be `grep.exs`, right?

Comment (cezarykosko) [inline exercises/grep/grep_test.exs]: hahah, obviously, "I'm gonna fix that after I've written the tests", duh... sorry :(

**Comment by devonestes** [comment]
Nice, all looks good - thanks for the contribution! 👍

**Pull request merged**